apps/examples/heap_stat_realtime/heap_stat_realtime.c

```diff
@@ -0,0 +1,71 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file heap_stat_realtime.c
+
+/// @brief Show the stat of heap for realtime support such as the number of misses of memory allocation within a constant time.
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <tinyara/mm/mm.h>
+#include <stdio.h>
+
+static int heap_stat_realtime(int argc, char *argv[])
+{
+	char *addr;
+	int ndx;
+	size_t miss[MM_REALTIME_SUPPORT_NUMOF_SIZES] = { 0, };
+	int num[MM_REALTIME_SUPPORT_NUMOF_SIZES] = { 0, };
+
+	addr = (char *)malloc(sizeof(char));
+
+	if (argc > 2) {
```

Comment:
Here, my feedback.
1. Null check is necessary, but this mem alloc is intended to get heap stat.
2. I'll do when I need to extend this function further.
3. if argc <= 2, 'heapstat' just goes well.

Reply:
I mean if user type heapstat reset blah blah blah it also works. is it intended? of course some os accepts it but some other doesn't accept it. I was just curious because you added usaage below..

Reply:
Yes, it was intentionally allowed to keep the code simple. Thanks for your comment, though.